pandas/core/indexing.py

```diff
@@ -1649,8 +1649,10 @@ def _setitem_with_indexer_split_path(self, indexer, value, name: str):
         if isinstance(indexer[0], np.ndarray) and indexer[0].ndim > 2:
             raise ValueError(r"Cannot set values with ndim > 2")
 
-        if isinstance(value, ABCSeries) and name != "iloc":
-            value = self._align_series(indexer, value)
+        if (isinstance(value, ABCSeries) and name != "iloc") or isinstance(value, dict):
```

Comment:
maybe put an if clause before this
```
if isinstance(value, dict):
    value = Series(value)
```

and leave the original, as a bit more clear? (e.g. you know i like terser code but this compound and is treating very differnt cases); also we are boxing an already created Series with Series (nbd but not as clean)

Reply:
the trouble with that is then it wouldn't do the alignment if name == "iloc"